### PR TITLE
Added swagger support

### PR DESCRIPTION
**Pull request by ValentinTyhonov**
- Added swagger for publish service
- Changed year in copyright headers from 2017 to 2018

### CODE REVIEW

File: publish-service/build.gradle [hunk 1]
```diff
@@ -79,5 +79,12 @@ dependencies {
     compileOnly("org.springframework.boot:spring-boot-starter-tomcat")
     providedRuntime("org.springframework.boot:spring-boot-starter-tomcat")
     testCompile("org.springframework.boot:spring-boot-starter-tomcat")
-        
+
+    //gson support
+    compile('com.google.code.gson:gson:2.6.2')
```
Comment (d-olausson-ericsson) [inline publish-service/build.gradle]: Can you make it consistent with the rest of the compile statement? Either we have compile("...") or compile "..."

File: publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java [hunk 2]
```diff
@@ -100,35 +111,41 @@ public ResponseEntity send(@RequestParam(value = "mp", required = false) String
...
      * @param tag
      *            (not required)
-     * @param rk
+     * @param routingKey
      *            (not required)
      * @return A response entity which contains http status and result
-     * @exception IOException
-     *                On input error.
-     * @see IOException
+//     * @exception IOException
```
Comment (d-olausson-ericsson) [inline publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java]: remove commented lines?

File: publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java [hunk 3]
```diff
@@ -78,12 +83,18 @@ public void setRestTemplate(RestTemplate restTemplate) {
...
+            @ApiResponse(code = 400, message = "Invalid event content"),
+            @ApiResponse(code = 404, message = "RabbitMq properties not found"),
+            @ApiResponse(code = 500, message = "Internal server error"),
+            @ApiResponse(code = 503, message = "Service Unavailable") })
     @RequestMapping(value = "/producer/msg", method = RequestMethod.POST)
     @ResponseBody
-    public ResponseEntity send(@RequestParam(value = "mp", required = false) String msgProtocol,
-            @RequestParam(value = "ud", required = false) String userDomain,
+    public ResponseEntity send(@RequestParam(value = "msgProtocol", required = false) String msgProtocol,
+            @RequestParam(value = "userDomain", required = false) String userDomain,
```
Comment (xvinosi-github) [inline publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java]: its not a good idea to change the reuest parameter names now as they have been alreasy in use from soem time.

**Comment by ValentinTyhonov** [comment]
Sorry. My fault. Reverted to Raja's commit and raised new pull request.

**Pull request merged**